tests/unit/Rules/EvenTest.php

```diff
@@ -13,60 +13,46 @@
 
 namespace Respect\Validation\Rules;
 
-use PHPUnit\Framework\TestCase;
+use Respect\Validation\Test\RuleTestCase;
 
 /**
  * @group  rule
  * @covers \Respect\Validation\Rules\Even
  * @covers \Respect\Validation\Exceptions\EvenException
+ *
+ * @author Jean Pimentel <jeanfap@gmail.com>
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ * @author Paul karikari <paulkarikari1@gmail.com>
  */
-class EvenTest extends TestCase
+class EvenTest extends RuleTestCase
 {
-    protected $evenValidator;
-
-    protected function setUp(): void
-    {
-        $this->evenValidator = new Even();
-    }
-
-    /**
-     * @dataProvider providerForEven
-     */
-    public function testEvenNumbersShouldPass($input): void
-    {
-        self::assertTrue($this->evenValidator->validate($input));
-        $this->evenValidator->check($input);
-        $this->evenValidator->assert($input);
-    }
-
     /**
-     * @dataProvider providerForNotEven
+     * @dataProvider providerForInvalidInput
      * @expectedException \Respect\Validation\Exceptions\EvenException
      */
-    public function testNotEvenNumbersShouldFail($input): void
+    public function testNotEvenNumbersShouldThrowException($validator, $input): void
     {
-        self::assertFalse($this->evenValidator->validate($input));
-        $this->evenValidator->assert($input);
+        $validator->assert($input);
     }
 
-    public function providerForEven()
+    public function providerForValidInput(): array
     {
         return [
-            [''],
-            [-2],
-            [-0],
-            [0],
-            [32],
+            [new Even(), ''],
```

Comment:
That's actually bad, so as it looks like we have a bug! 

Could you also change the `Even` rule by adding a verification if the input is numeric before checking if it's an even number? (`is_numeric($input)`)